web/ts/edit-course.ts

```diff
@@ -620,44 +727,93 @@ export function createLectureForm(args: { s: [] }) {
                     payload.duration = 0; // premieres have no explicit end set -> use "0" here
                 }
                 postData("/api/course/" + this.courseID + "/createLecture", payload)
-                    .then(() => {
-                        this.loading = false;
+                    .then(async (res) => {
+                        const { ids } = await res.json();
+                        const url = new URL(window.location.href);
+                        url.hash = `lectures:${ids.join(",")}`;
+                        window.location.assign(url);
                         window.location.reload();
                     })
-                    .catch(() => {
+                    .catch((e) => {
+                        console.log(e);
+                        this.currentTab = currentTab;
                         this.loading = false;
                         this.error = true;
                     });
             }
         },
-        uploadVod() {
-            const xhr = new XMLHttpRequest();
-            const vodUploadFormData = new FormData();
-            vodUploadFormData.append("file", this.formData.file[0]);
-            xhr.onloadend = () => {
-                if (xhr.status === 200) {
-                    window.location.reload();
-                } else {
-                    this.error = true;
-                }
-            };
-            xhr.upload.onprogress = (e: ProgressEvent) => {
-                if (!e.lengthComputable) {
-                    return;
-                }
-                window.dispatchEvent(
-                    new CustomEvent("voduploadprogress", { detail: Math.floor(100 * (e.loaded / e.total)) }),
-                );
+
+        async uploadVod(): Promise<number> {
+            const uploadPres = this.formData.presFile[0] != null;
+            const uploadCam = this.formData.camFile[0] != null;
```

Comment:
COMB should not be required, the only requirement should be `at least one file is present`